Roles/Impostor/Murderer.cs

```diff
@@ -0,0 +1,129 @@
+    }
+    public override void Add(byte playerId)
+    {
+        playerIdList.Add(playerId);
+        if (playerId == PlayerControl.LocalPlayer.PlayerId && Main.nickName.Length != 0) MurdererVictim[playerId] = Main.nickName;
+        else MurdererVictim[playerId] = Utils.GetPlayerById(playerId).Data.PlayerName;
+
+        if (!AmongUsClient.Instance.AmHost) return;
+        if (!Main.ResetCamPlayerList.Contains(playerId))
+            Main.ResetCamPlayerList.Add(playerId);
+    }
+
+    private static void SendRPC(byte playerId)
+    {
+        MessageWriter writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId, (byte)CustomRPC.SyncRoleSkill, SendOption.Reliable, -1);
+        writer.WritePacked((int)CustomRoles.Murderer);
+        writer.Write(playerId);
+        AmongUsClient.Instance.FinishRpcImmediately(writer);
+    }
+
+    public static bool CheckMurdererVictim(byte playerId) => MurdererVictim.ContainsKey(playerId);
+
+    public override void SetKillCooldown(byte id) => Main.AllPlayerKillCooldown[id] = DefaultKillCooldown.GetFloat();
+    public override bool CanUseKillButton(PlayerControl pc) => true;
+
+    public static void RpcChangeSkin(PlayerControl pc, GameData.PlayerOutfit newOutfit, uint level)
+    {
+        var sender = CustomRpcSender.Create(name: $"Doppelganger.RpcChangeSkin({pc.Data.PlayerName})");
+
+        pc.SetName(newOutfit.PlayerName);
+        sender.AutoStartRpc(pc.NetId, (byte)RpcCalls.SetName)
+            .Write(newOutfit.PlayerName)
+        .EndRpc();
+
+        Main.AllPlayerNames[pc.PlayerId] = newOutfit.PlayerName;
+
+        pc.SetColor(newOutfit.ColorId);
+        sender.AutoStartRpc(pc.NetId, (byte)RpcCalls.SetColor)
+            .Write(newOutfit.ColorId)
+        .EndRpc();
+
+        pc.SetHat(newOutfit.HatId, newOutfit.ColorId);
+        sender.AutoStartRpc(pc.NetId, (byte)RpcCalls.SetHatStr)
+            .Write(newOutfit.HatId)
+        .EndRpc();
+
+        pc.SetSkin(newOutfit.SkinId, newOutfit.ColorId);
+        sender.AutoStartRpc(pc.NetId, (byte)RpcCalls.SetSkinStr)
+            .Write(newOutfit.SkinId)
+        .EndRpc();
+
+        pc.SetVisor(newOutfit.VisorId, newOutfit.ColorId);
+        sender.AutoStartRpc(pc.NetId, (byte)RpcCalls.SetVisorStr)
+            .Write(newOutfit.VisorId)
+        .EndRpc();
+
+        pc.SetNamePlate(newOutfit.NamePlateId);
+        sender.AutoStartRpc(pc.NetId, (byte)RpcCalls.SetNamePlateStr)
+            .Write(newOutfit.NamePlateId)
+            .EndRpc();
+        
+        pc.SetLevel(level);
+        sender.AutoStartRpc(pc.NetId, (byte)RpcCalls.SetLevel)
+            .Write(level)
+            .EndRpc();
+
+        sender.SendMessage();
+    }
+
+    public override bool OnCheckMurderAsKiller(PlayerControl killer, PlayerControl target)
+    {
+        if (killer == null || target == null) return false;
+
+        MurdererVictim[target.PlayerId] = target.GetRealName();
+        RpcChangeSkin(target, UnidentifiableOutfit, 0);
```

Comment:
@ryuk201198 so murderer changes the target skin, while butcherer kills the target mutiple times. Tough I don't know how much this is helpful other than maybe bait.